posix/unix.c

```diff
@@ -218,20 +218,26 @@ static unixsock_t *unixsock_get(unsigned id)
 }
 
 
-static void unixsock_put(unixsock_t *r)
+static void unixsock_put(unixsock_t *s)
 {
 	proc_lockSet(&unix_common.lock);
-	if (--r->refs <= 0) {
-		lib_rbRemove(&unix_common.tree, &r->linkage);
+	if (--s->refs <= 0) {
+		lib_rbRemove(&unix_common.tree, &s->linkage);
+
+		/* TODO: handle connecting socket */
+
+		if (s->connect != NULL && !(s->state & US_DISCONNECTED))
+			s->connect->state |= US_DISCONNECTED;
```

Comment:
maybe we should set s->connect->connect = NULL  to avoid dangling pointers?

Reply:
Yeah, dangling pointer is not nice.